Library/Homebrew/attestation.rb

```diff
@@ -103,8 +103,22 @@ def self.check_attestation(bottle, signing_repo, signing_workflow = nil, subject
       # for all attestations that match the input's digest. We want to additionally
       # filter these down to just the attestation whose subject matches the bottle's name.
       subject = bottle.filename.to_s if subject.blank?
-      attestation = attestations.find do |a|
-        a.dig("verificationResult", "statement", "subject", 0, "name") == subject
+
+      attestation = if bottle.tag.to_sym == :all
+        # :all-tagged bottles are created by `brew bottle --merge`, and are not directly
+        # bound to their own filename (since they're created by deduplicating other filenames).
+        # To verify these, we parse each attestation subject and look for one with a matching
+        # formula (name, version), but not an exact tag match.
+        # This is sound insofar as the signature has already been verified. However,
+        # longer term, we should also directly attest to `:all`-tagged bottles.
+        attestations.find do |a|
+          actual_subject = a.dig("verificationResult", "statement", "subject", 0, "name")
+          actual_subject.start_with? "#{bottle.filename.name}--#{bottle.filename.version}"
+        end
+      else
+        attestations.find do |a|
+          a.dig("verificationResult", "statement", "subject", 0, "name") == subject
```

Comment:
**nitpick (non-blocking):** This dig _could_ be a function, e.g.

```ruby
def get_result_subject_name(attestation)
  attestation.dig("verificationResult", "statement", "subject", 0, "name")
end
```

so there's one place that extracts the info; one place this path is constructed in its current implementation.

NBD, though.

Reply:
Actually, this is going to cause a decent bit of test churn, since factoring out into a helper will require updating the expected calls in each spec. So I'm going to punt on this for a follow-up 🙂